src/backend/tcop/postgres.c

```
@@ -1638,13 +1638,6 @@ restore_guc_to_QE(void )
 			 * we can not keep alive gang anymore.
 			 */
 			DisconnectAndDestroyAllGangs(false);
-			/*
-			 * when qe elog an error, qd will use ReThrowError to
-			 * re throw the error, the errordata_stack_depth will ++,
-			 * when we catch the error we should reset errordata_stack_depth
-			 * by FlushErrorState.
-			 */
-			FlushErrorState();
```

Why we remove `FlushErrorState()` here?

---

When I did a global search for `ReThrowError` in the codebase, I found this comment. 
https://github.com/greenplum-db/gpdb/commit/c4f03b876030b40a4eda17b409c91a6d81288958#diff-0e44d172203e2969b2440b26d5a272c39e3dcef3013351e3357c32a97a7b7944
Referring to this commit, according to its logic, it seems that because `ReThrowError` was called in the previous code, this PR changed it to `ThrowErrorData`, and therefore `FlushErrorState` is no longer needed.

---

Ah, I see. Generally speaking, in a `PG_CATCH()` block, no matter the error is thrown by `ThrowErrorData()`/`ereport()`/`ReThrowError()`, we should either comsuming the error by `FlushErrorState()` or throw the error to its upper caller by `PG_RE_THROW()`/`ReThrowError()`. There indeed are some special cases that need to preserve the error context and don't call any of these two functions. But this one is not one of those special cases.

In this case, we want to ignore the error, so we use `FlushErrorState()` to consume the error. If we remove `FlushErrorState()` here, the error data stack keeps growing and will exceed the limit `ERRORDATA_STACK_SIZE(10)` and the server will panic.

---

OK, I see what you mean, calling `FlushErrorState` in `PG_CATCH` can prevent the excessive growth of errorstack